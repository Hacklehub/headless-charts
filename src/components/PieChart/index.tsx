import { PieArcDatum, arc, pie } from 'd3';
import { pointer, select, selectAll } from 'd3-selection';

import { interpolate } from 'd3-interpolate';
import { mergeTailwindClasses } from '../../utils';
import { min } from 'd3-array';
import { useEffect } from 'react';

/* eslint-disable @typescript-eslint/no-explicit-any */

interface DataItem {
  name: string;
  [key: string]: any;
}

interface ClassNamePoints {
  classMap: { [key: string]: string };
}

interface DrawingOptions {
  duration: number;
}

interface TooltipOptions {
  html?: (d: any) => string;
  className?: string;
  keys?: Array<PieArcDatum<DataItem> | any>;
}

interface LabelOptions {
  radius?: number;
  key: string;
  text?: (data: DataItem) => string;
  className?: string;
  labelsMap?: { [key: string]: string };
}

interface PieChartProps {
  data: DataItem[];
  id: string;
  className?: string;
  classNamePoints?: ClassNamePoints;
  paddingBar?: number;
  padding?: {
    left: number;
    right?: number;
    top: number;
    bottom?: number;
  };
  paddingAngle?: number;
  cornerRadius?: number;
  margin?: {
    left: number;
    right: number;
    top: number;
    bottom: number;
  };
  innerRadius?: number;
  value: string;
  drawing?: DrawingOptions;
  tooltip?: TooltipOptions;
  labels: LabelOptions;
}

const PieChart = ({
  data,
  id,
  className = '',
  classNamePoints = { classMap: {} },
  padding = {
    left: 0,
    top: 0,
  },
  paddingAngle = 0,
  cornerRadius = 0,
  margin = {
    left: 40,
    right: 40,
    top: 40,
    bottom: 40,
  },
  innerRadius = 0,
  value,
  drawing,
  tooltip,
  labels,
}: PieChartProps) => {
  const refreshChart = () => {
    const svg = select(`#${id}`);
    svg.selectAll('*').remove();

    const width = +svg.style('width').split('px')[0];
    const height = +svg.style('height').split('px')[0];

    const g = svg.append('g');

    const pieFn = pie<DataItem>()
      .sort(null)
      .value((d: { [x: string]: any }) => d[value])
      .padAngle(paddingAngle);

    const chartArea = [
      width - margin.left - margin.right,
      height - margin.top - margin.bottom,
    ];

    const radius = min(chartArea.map((a) => a / 2)) || 0;

    const arcFn = arc()
      .innerRadius(radius * innerRadius)
      .outerRadius(radius)
      .padAngle(paddingAngle)
      .cornerRadius(cornerRadius);

    const labelArc =
      labels?.radius &&
      arc<DataItem>()
        .innerRadius(radius * labels.radius)
        .outerRadius(radius * labels.radius);

    const arcs = pieFn(data);

    const pathsG = g
      .append('g')
      .attr(
        'transform',
        `translate(${padding.left + margin.left + chartArea[0] / 2},${
          margin.top + padding.top + chartArea[1] / 2
        })`
      );

    const paths = pathsG
      .selectAll('path')
      .data(arcs)
      .join('path')
      .attr(
        'class',
        (d) =>
          `${
            classNamePoints.classMap[d.data.name] || ''
          } fill-current stroke-current`
      )
      /* eslint-disable */
      // @ts-ignore
      .attr('d', arcFn)
      .on('mouseenter', function (event, d) {
        if (tooltip) {
          tooltipDiv.style('opacity', '1');
          const [bX, bY] = pointer(event, select('body'));
          tooltipDiv.style('left', `${bX + 10}px`).style('top', `${bY + 10}px`);
          tooltipDiv.html(
            tooltip && tooltip.html
              ? tooltip.html(d)
              : 
              tooltip.keys
              ? 
                tooltip.keys
                  // @ts-ignore
                  .map((key) => `${key}: ${d[key] || ''}`)
                  .join('<br/>')
              : `${d.data.name} = ${d.data[value]} `
          );
        }
      })
      .on('mousemove', function (event) {
        const [bX, bY] = pointer(event, select('body'));
        tooltipDiv.style('left', `${bX + 10}px`).style('top', `${bY + 10}px`);
      })
      .on('mouseleave', function () {
        tooltip &&
          tooltipDiv
            .style('opacity', '0')
            .style('left', '0px')
            .style('top', '0px');
      });

    drawing &&
      drawing.duration &&
      paths
        .transition()
        .duration(drawing.duration)
        .attrTween('d', function (d) {
          const i = interpolate({ startAngle: 0, endAngle: 0 }, d);

          // @ts-ignore
          return (t) => arcFn(i(t));
        });

    const labelsG = labelArc && pathsG.append('g').attr('class', 'labels');

    labelArc &&
      labelsG &&
      labelsG
        .selectAll('g')
        .data(arcs)
        .enter()
        .append('text')
        .attr(
          'transform',
          // @ts-ignore
          (d: { data: DataItem }) => `translate(${labelArc.centroid(d)})`
        )
        .attr('text-anchor', 'middle')
        .attr(
          'class',
          `${labels?.className || ''} ${
            (labels.labelsMap && labels.labelsMap[labels.key]) || ''
          } fill-current`
        )
        .text((d: { data: DataItem }) =>
          labels.text ? labels.text(d.data) : d.data[labels.key]
        );
    /* eslint-enable */
    const tooltipDiv = select('#root')
      .append('div')
      .attr('id', 'tooltip')
      .style('position', 'absolute')
      .style('opacity', '0')
      .attr('class', `tooltip ${(tooltip && tooltip.className) || ''}`);
  };

  useEffect(() => {
    refreshChart();
    return () => {
      selectAll('.tooltip').remove();
    };
  }, [data]);

  return (
    <svg
      id={id}
      className={mergeTailwindClasses(
        `w-full md:w-6/12 lg:w-4/12 dark:bg-gray-800 text-gray-900 dark:text-gray-50 chart h-80`,
        className || ''
      )}
    />
  );
};

export default PieChart;
