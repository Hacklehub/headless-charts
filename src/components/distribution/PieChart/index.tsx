import { arc, pie } from 'd3';
import { pointer, select } from 'd3-selection';
import { useCallback, useEffect } from 'react';

import { defaultChartClassNames } from '../../../utils';
import { interpolate } from 'd3-interpolate';
import { mergeTailwindClasses } from '../../../utils';
import { min } from 'd3-array';

/* eslint-disable @typescript-eslint/no-explicit-any */

interface DataItem {
  name: string;
  [key: string]: any;
}

interface ClassNamePoints {
  classMap: { [key: string]: string };
}

interface DrawingOptions {
  duration: number;
}

interface LabelOptions {
  radius?: number;
  key: string;
  text?: (data: DataItem) => string;
  className?: string;
  labelsMap?: { [key: string]: string };
}

interface PieChartProps {
  data: DataItem[];
  id: string;
  className?: string;
  classNamePoints?: ClassNamePoints;
  paddingBar?: number;
  padding?: {
    left: number;
    right?: number;
    top: number;
    bottom?: number;
  };
  paddingAngle?: number;
  cornerRadius?: number;
  margin?: {
    left: number;
    right: number;
    top: number;
    bottom: number;
  };
  startAngle?: number;
  endAngle?: number;
  innerRadius?: number;
  value: string;
  drawing?: DrawingOptions;
  tooltip?: {
    className?: string;
    html?: (d: any) => string;
    keys?: string[];
  };
  labels: LabelOptions;
}

const PieChart = ({
  data,
  id,
  className = '',
  classNamePoints = { classMap: {} },
  padding = {
    left: 0,
    top: 0,
  },
  paddingAngle = 1,
  startAngle = 0,
  endAngle = 360,
  cornerRadius = 0,
  margin = {
    left: 40,
    right: 40,
    top: 40,
    bottom: 40,
  },
  innerRadius = 0,
  value,
  drawing,
  tooltip,
  labels,
}: PieChartProps) => {
  const refreshChart = useCallback(() => {
    const svg = select(`#${id}`);
    svg.selectAll('*').remove();

    const width = +svg.style('width').split('px')[0];
    const height = +svg.style('height').split('px')[0];

    const g = svg.append('g');

    const pieFn = pie<DataItem>()
      .value((d: { [x: string]: any }) => d[value])
      .startAngle((startAngle / 180) * Math.PI || 0)
      .endAngle((endAngle / 180) * Math.PI || 2 * Math.PI)
      .padAngle(paddingAngle);

    const chartArea = [
      width - margin.left - margin.right,
      height - margin.top - margin.bottom,
    ];

    const radius =
      min(
        endAngle - startAngle <= 180 ? chartArea : chartArea.map((a) => a / 2)
      ) || 0;

    const arcFn = arc()
      .innerRadius(radius * innerRadius)
      .outerRadius(radius)
      .padAngle((paddingAngle / 360) * (2 * Math.PI))
      .cornerRadius(cornerRadius);

    const labelArc =
      labels?.radius &&
      arc<DataItem>()
        .innerRadius(radius * labels.radius)
        .outerRadius(radius * labels.radius);

    const arcs = pieFn(data);

    const pathsG = g
      .append('g')
      .attr(
        'transform',
        `translate(${padding.left + margin.left + chartArea[0] / 2},${
          endAngle - startAngle <= 180
            ? height - margin.bottom - (padding.bottom || 0)
            : margin.top + padding.top + chartArea[1] / 2
        })`
      );
    const tooltipDiv = select('body')
      .append('div')
      .attr('id', 'tooltip')
      .style('position', 'absolute')
      .style('opacity', '0')
      .attr('class', `${tooltip?.className || ''}`);

    const paths = pathsG
      .selectAll('path')
      .data(arcs)
      .join('path')
      .attr('class', (d) =>
        mergeTailwindClasses(
          classNamePoints.classMap[d.data.name],
          'fill-current stroke-current'
        )
      )
      /* eslint-disable */
      // @ts-ignore
      .attr('d', arcFn)
      .on('mouseenter', function (event: MouseEvent, d: any) {
        if (tooltip) {
          tooltipDiv.style('opacity', 1);
          const [bX, bY] = pointer(event, select('body'));
          tooltipDiv.style('left', `${bX + 10}px`).style('top', `${bY + 10}px`);
          tooltipDiv.html(
            tooltip && tooltip.html
              ? tooltip.html(d)
              : tooltip.keys
              ? tooltip.keys
                  .map((key) => `${key}: ${d.data[key] || ''}`)
                  .join('<br/>')
              : `${d.data.name} = ${d.data[value]} `
          );
        }
      })
      .on('mousemove', function (event: MouseEvent) {
        const [bX, bY] = pointer(event, select('body'));
        tooltipDiv.style('left', `${bX + 10}px`).style('top', `${bY + 10}px`);
      })
      .on('mouseleave', function () {
        tooltip &&
          tooltipDiv
            .style('opacity', '0')
            .style('left', '0px')
            .style('top', '0px');
      });

    drawing &&
      drawing.duration &&
      paths
        .transition()
        .duration(drawing.duration)
        .attrTween('d', function (d) {
          const i = interpolate({ startAngle: 0, endAngle: 0 }, d);

          // @ts-ignore
          return (t) => arcFn(i(t));
        });

    const labelsG = labelArc && pathsG.append('g').attr('class', 'labels');

    labelArc &&
      labelsG &&
      labelsG
        .selectAll('g')
        .data(arcs)
        .enter()
        .append('text')
        .attr(
          'transform',
          // @ts-ignore
          (d: { data: DataItem }) => `translate(${labelArc.centroid(d)})`
        )
        .attr('text-anchor', 'middle')
        .attr(
          'class',
          `${labels?.className || ''} ${
            (labels.labelsMap && labels.labelsMap[labels.key]) || ''
          } fill-current`
        )
        .text((d: { data: DataItem }) =>
          labels.text ? labels.text(d.data) : d.data[labels.key]
        );
    /* eslint-enable */
  }, [
    id,
    startAngle,
    endAngle,
    paddingAngle,
    margin,
    innerRadius,
    cornerRadius,
    labels,
    data,
    padding,
    tooltip,
    drawing,
    value,
    classNamePoints,
  ]);

  useEffect(() => {
    refreshChart();
  }, [data, refreshChart]);

  return (
    <svg
      id={id}
      className={mergeTailwindClasses(defaultChartClassNames, className)}
    />
  );
};

export default PieChart;
