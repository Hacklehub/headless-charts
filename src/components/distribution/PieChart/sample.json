[
  {
    "name": "Product A",
    "USA": 1000,
    "Europe": 4000,
    "APAC": 5000,
    "Africa": 1000
  },
  { "name": "Product B", "USA": 36000, "Europe": 9000, "APAC": 4000 },
  { "name": "Product C", "USA": 6000, "Europe": 4000, "APAC": 1000 }
]
