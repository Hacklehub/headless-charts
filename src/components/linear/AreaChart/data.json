[
  {
    "year": 2012,
    "macbook": 23.22,
    "services": 12.89,
    "wearables": 10.76,
    "ipad": 30.94,
    "iphone": 78.69
  },
  {
    "year": 2013,
    "macbook": 21.48,
    "services": 16.05,
    "wearables": 10.12,
    "ipad": 31.98,
    "iphone": 91.28
  },
  {
    "year": 2014,
    "macbook": 24.08,
    "services": 18.06,
    "wearables": 8.379,
    "ipad": 30.28,
    "iphone": 101.99
  },
  {
    "year": 2015,
    "macbook": 25.47,
    "services": 19.91,
    "wearables": 10.07,
    "ipad": 23.23,
    "iphone": 155.04
  },
  {
    "year": 2016,
    "macbook": 22.83,
    "services": 24.35,
    "wearables": 11.13,
    "ipad": 20.63,
    "iphone": 136.7
  },
  {
    "year": 2017,
    "macbook": 25.57,
    "services": 32.7,
    "wearables": 12.83,
    "ipad": 18.8,
    "iphone": 139.34
  },

  {
    "year": 2018,
    "macbook": 25.2,
    "services": 39.75,
    "wearables": 17.38,
    "ipad": 18.38,
    "iphone": 164.89
  },
  {
    "year": 2019,
    "macbook": 25.74,
    "services": 46.29,
    "wearables": 24.48,
    "ipad": 21.28,
    "iphone": 142.38
  },
  {
    "year": 2020,
    "macbook": 28.62,
    "services": 53.77,
    "wearables": 30.62,
    "ipad": 23.72,
    "iphone": 137.78
  },

  {
    "year": 2021,
    "macbook": 35.19,
    "services": 68.42,
    "wearables": 38.37,
    "ipad": 31.86,
    "iphone": 191.97
  },
  {
    "year": 2022,
    "macbook": 40.18,
    "services": 78.13,
    "wearables": 41.24,
    "ipad": 29.29,
    "iphone": 205.49
  }
]
