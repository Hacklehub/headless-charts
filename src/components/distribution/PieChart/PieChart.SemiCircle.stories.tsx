import { Meta, StoryObj } from '@storybook/react';

import PieChart from '.';
import data from './sample.json';

/**
 * SemiCircle charts are specifically useful for certain representations like political polls or seat shares, since they look like the seating arrangement in a parliament.
 */
const meta: Meta<typeof PieChart> = {
  title: 'Distribution/PieChart/SemiCircle',
  component: PieChart,
  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof PieChart>;

const classNameMap = {
  'Product A': 'fill-purple-700',
  'Product B': 'fill-purple-500',
  'Product C': 'fill-purple-300',
};

/**
 * SemiCircle charts are a variant of pie charts. Simply specify `startAngle` and `endAngle` props and set it to -90 and 90.
 */
export const SemiCircle: Story = {
  args: {
    id: 'tooltip',
    data,
    valueKey: 'USA',
    classNameMap,
    startAngle: -90,
    endAngle: 90,
  },
};

/**
 * Drawing the chart can be animated by specifying a `duration` in milliseconds.
 */

export const Drawing: Story = {
  args: {
    ...SemiCircle.args,
    id: 'drawing-pie-chart',
    drawing: {
      duration: 1000,
    },
  },
};
