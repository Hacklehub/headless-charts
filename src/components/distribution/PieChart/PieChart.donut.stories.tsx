import { Meta, StoryObj } from '@storybook/react';

import PieChart from '.';
import data from './sample.json';

/**
 * Donut charts are a variant of pie charts. Simply specify an `innerRadius` prop
 */
const meta: Meta<typeof PieChart> = {
  title: 'Distribution/PieChart/Donuts',
  component: PieChart,
  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof PieChart>;

const classNameMap = {
  'Product A': 'fill-purple-700',
  'Product B': 'fill-purple-500',
  'Product C': 'fill-purple-300',
};

/**
 * From the normally styled `PieChart` component, simply specify an `innerRadius` prop.
 */
export const Donut: Story = {
  args: {
    id: 'donut',
    data,
    valueKey: 'USA',
    classNameMap,
    innerRadius: 0.65,
  },
};

/**
 * The `labels` prop can be used to add labels to the slices.
 * */

export const Labelled: Story = {
  args: {
    ...Donut.args,
    id: 'labelled-donut',
    labels: {
      radius: 1.2,
    },
  },
};

/**
 * The `drawing` prop can be used to animate the drawing of the slices.
 */

export const Drawing: Story = {
  args: {
    ...Donut.args,
    id: 'drawing-donut',
    drawing: {
      duration: 1000,
    },
  },
};

/**
 * Props like `drawing` and `labels` can be used together.
 */
export const DrawingWithLabels: Story = {
  args: {
    id: 'drawing-with-labels-donut',
    ...Labelled.args,
    ...Drawing.args,
  },
};

/**
 * The paddingAngle prop is used to add padding between slices (in degrees).
 */
export const PaddingAngle: Story = {
  args: {
    ...Drawing.args,
    id: 'padding-angle-donut',
    paddingAngle: 1,
  },
};

/**
 * The cornerRadius prop is used to add rounded corners to the slices.
 */
export const CornerRadius: Story = {
  args: {
    ...Drawing.args,
    id: 'corner-radius-donut',
    cornerRadius: 4,
  },
};

