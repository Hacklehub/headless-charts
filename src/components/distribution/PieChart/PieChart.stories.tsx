/* eslint-disable @typescript-eslint/no-explicit-any */
import { Meta, StoryObj } from '@storybook/react';

import PieChart from '.';
import data from './sample.json';

/**
 * Pie charts can be used to show how much each category represents as part of a whole. They are useful for showing the distribution of a dataset.
 * 
 */
export default {
  title: 'Distribution/PieChart',
  component: PieChart,
  tags: ['autodocs'],
} as Meta;

type Story = StoryObj<typeof PieChart>;

const classNameMap = {
  'Product A': 'fill-purple-700 text-red-700',
  'Product B': 'fill-purple-500 ',
  'Product C': 'fill-purple-300 ',
};

/**
* The default chart will iterate through the `data` prop and takes the `valueKey` prop as the value to be represented, and `nameKey` as the name of the category.
*/
export const Default: Story = {
  args: {
    data,
    id: 'default-pie-chart',
    valueKey: 'USA',
    nameKey: 'name',
  },
};

/**
 * However, the default chart will not be styled. You can provide a `classNameMap` prop, with a list of possible values for the `nameKey` prop.
 */
export const Styled: Story = {
  args: {
    ...Default.args,
    id: 'styled-pie-chart',
    classNameMap,
  },
};

/**
 * Animating the chart can be animated by specifying a `duration` in milliseconds in `drawing` prop.
 */
export const Drawing: Story = {
  args: {
    ...Styled.args,
    id: 'drawing-pie-chart',
    drawing: {
      duration: 1000,
    },
  },
};

/**
 * You can add labels to the chart by specifying a `labels` prop. The `labels` prop takes a `radius` prop, which is the radius of the circle where the labels will be placed. The `key` prop is the name of the category. The `text` prop is a function that takes the data and returns the text to be displayed. The `className` prop is a string of tailwind classes to be applied to the labels.
 */

export const Labelled: Story = {
  args: {
    ...Styled.args,
    id: 'labelled-pie-chart',
    labels: {
      radius: 1.2,
      className: 'p-2 rounded',
      classNameMap,
    },
  },
};

/**
 * You can add the `startAngle` prop to rotate the chart.
 */
export const StartAngle90: Story = {
  args: {
    ...Styled.args,
    id: 'pie-chart-start-angle',
    startAngle: 90,
  },
};

export const PaddingAngle: Story = {
  args: {
    ...Default.args,
    id: 'pie-chart-padding-angle',
    paddingAngle: 1,
  },
};

export const CornerRadius: Story = {
  args: {
    ...PaddingAngle.args,
    id: 'pie-chart-corner-radius',
    cornerRadius: 5,
  },
};
