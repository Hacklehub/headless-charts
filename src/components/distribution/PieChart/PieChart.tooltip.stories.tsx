/* eslint-disable @typescript-eslint/no-explicit-any */
import { Meta, StoryObj } from '@storybook/react';

import PieChart from '.';
import data from './sample.json';

/**
 * Donut charts are a variant of pie charts. Simply specify an `innerRadius` prop
 */
const meta: Meta<typeof PieChart> = {
  title: 'Distribution/PieChart/Tooltips',
  component: PieChart,
  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof PieChart>;

const classNameMap = {
  'Product A': 'fill-purple-700',
  'Product B': 'fill-purple-500',
  'Product C': 'fill-purple-300',
};

/**
 * Tooltips can be enabled by adding a `tooltip` prop (`{}` will render default behaviour).
 */
export const Tooltip: Story = {
  args: {
    id: 'tooltip',
    data,
    valueKey: 'USA',
    classNameMap,
    tooltip: {},
  },
};

/**
 * Tooltip divs can be styled by passing a `tooltip` prop with a `className` property.
 * */
export const TooltipStyled: Story = {
  args: {
    ...Tooltip.args,
    id: 'tooltip-styled',
    tooltip: {
      className: 'bg-purple-800 text-white p-2 rounded',
    },
  },
};

/**
 * The `tooltip.keys` prop can be used to specify which keys in the data array to show in the tooltip (Even those not used in creating the pie).
 * */
export const TooltipCustom: Story = {
  args: {
    ...Tooltip.args,
    id: 'tooltip-custom',
    tooltip: {
      keys: ['name', 'USA'],
    },
  },
};

/**
 * The `tooltip.html` prop can be used to specify a custom text to show in the tooltip. The function will be passed the data object.
 * */
export const TooltipCustomHtml: Story = {
  args: {
    ...Tooltip.args,
    id: 'tooltip-custom-html',
    tooltip: {
      className: 'bg-purple-800 text-white p-2 rounded',
      html: (d: any) =>
        `${d.data.name} sold ${d.value} in USA and ${d.data['Europe']} in Europe`,
    },
  },
};
