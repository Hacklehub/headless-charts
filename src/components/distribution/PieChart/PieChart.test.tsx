import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';

import PieChart from '.';
import data from './sample.json';

describe('PieChart', () => {
  it('renders PieChart component for snapshot test', () => {
    render(<PieChart
      data={data}
      id='pie-chart'
      valueKey='USA'
      nameKey='name' />);
    const pieChart = screen.getByTestId('pie-chart');
    expect(pieChart).toMatchSnapshot();
  });

  it('renders PieChart component', () => {
    render(<PieChart
      data={data}
      id='pie-chart'
      valueKey='USA'
      nameKey='name' />);
    const pieChart = screen.getByTestId('pie-chart');
    expect(pieChart).toBeInTheDocument();
  });

})