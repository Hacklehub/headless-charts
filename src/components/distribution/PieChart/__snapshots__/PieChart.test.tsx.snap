// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PieChart > renders PieChart component for snapshot test 1`] = `
<svg
  class="chart dark:bg-gray-800 dark:text-gray-50 h-64 lg:w-4/12 md:w-6/12 text-gray-900 w-full"
  data-testid="pie-chart"
  id="pie-chart"
>
  <g>
    <g
      transform="translate(0,0)"
    >
      <path
        class="fill-black"
        d="M0,0Z"
        data-testid="Product A"
        id="Product A"
      />
      <path
        class="fill-black"
        d="M0,0Z"
        data-testid="Product B"
        id="Product B"
      />
      <path
        class="fill-black"
        d="M0,0Z"
        data-testid="Product C"
        id="Product C"
      />
    </g>
  </g>
</svg>
`;
